Python/compile.c

```diff
@@ -7746,6 +7747,109 @@ assemble_jump_offsets(struct assembler *a, struct compiler *c)
     } while (extended_arg_recompile);
 }
 
+
+// Ensure each basicblock is only put onto the stack once.
+#define MAYBE_PUSH(B) do {                          \
+        if ((B)->b_visited == 0) {                  \
+            *(*stack_top)++ = (B);                  \
+            (B)->b_visited = 1;                     \
+        }                                           \
+    } while (0)
+
+static void
+scan_block_for_local(int target, basicblock *b, bool unsafe_to_start,
+                     basicblock ***stack_top)
+{
+    bool unsafe = unsafe_to_start;
+    for (int i = 0; i < b->b_iused; i++) {
+        struct instr *instr = &b->b_instr[i];
+        assert(instr->i_opcode != EXTENDED_ARG);
+        assert(instr->i_opcode != EXTENDED_ARG_QUICK);
+        assert(instr->i_opcode != LOAD_FAST__LOAD_FAST);
+        assert(instr->i_opcode != STORE_FAST__LOAD_FAST);
+        assert(instr->i_opcode != LOAD_CONST__LOAD_FAST);
+        assert(instr->i_opcode != STORE_FAST__STORE_FAST);
+        assert(instr->i_opcode != LOAD_FAST__LOAD_CONST);
```

Comment:
I recently added IS_ASSEMBLER_OPCODE to contain a list of opcodes that we don't emit in code gen stage, but rather the assembler can use them to replace other opcodes. Would the superinstruction opcodes fit in that list?

Reply:
As far as I know, super-instructions are only ever created at quickening time, not compile-time/assembly-time, so I'm not sure it's right to call them assembler opcodes. I think I remember someone hinting at that potentially changing so that all code is quickened by default? If this becomes the case then maybe it makes sense to add an `IS_SUPERINSTRUCTION()` macro?